sam/pairedSam.go

```diff
@@ -1 +1,122 @@
 package sam
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/exception"
+	"io"
+	"strings"
+)
+
+// SamPE is struct that contains 2 paired end Sam entries as well as a slice of Sam that represents all supplementary
+// alignments for the read-pair
+type SamPE struct {
+	A   Sam
+	B   Sam
+	Sup []Sam
+}
+
+// GoReadSamPeToChan takes a read-name sorted sam file and returns a chanel of SamPE and a Header
+func GoReadSamPeToChan(filename string) (<-chan SamPE, Header) {
+
+	//start by reading the file into a chanel of sam
+	data := make(chan Sam, 1000)
+	header := make(chan Header)
```

Comment:
When running this function that takes a single file as an input, would there ever be an instance where you would have more header?

Probably doesn't matter too much, but I was just wondering if something like
`var header Header` would be enough. Does `header := make(chan Header)` only make 1 goroutine? I forgot how that stuff works.

Reply:
Yeah its a good question. I did it that way because thats how it was done in the other sam readToChan functions. However, I also thought it was weird that make a channel for just the header. I look into this more tomorrow